src/segments/sitecore.go

```diff
@@ -2,77 +2,96 @@ package segments
 
 import (
 	"encoding/json"
-	"path/filepath"
-	"strings"
+	"path"
 
 	"github.com/jandedobbeleer/oh-my-posh/src/platform"
 	"github.com/jandedobbeleer/oh-my-posh/src/properties"
 )
 
+const (
+	sitecoreFileName = "sitecore.json"
+	sitecoreFolderName = ".sitecore"
+	userFileName = "user.json"
+	defaultEnpointName = "default"
+)
+
 type Sitecore struct {
-	props properties.Properties
-	env   platform.Environment
+    props   	properties.Properties
+    env     	platform.Environment
+
+    EndpointName 	string
+	CmHost 			string
+}
 
-	Environment string
-	Cloud       bool
+type EndpointConfig struct {
+	Host string `json:"host"`
 }
 
-type SitecoreConfig struct {
-	Endpoints struct {
-		Default struct {
-			Ref        string `json:"ref"`
-			AllowWrite bool   `json:"allowWrite"`
-			Host       string `json:"host"`
-			Variables  struct {
-			} `json:"variables"`
-		} `json:"default"`
-	} `json:"endpoints"`
+type UserConfig struct {
+	DefaultEndpoint string 						`json:"defaultEndpoint"`
+	Endpoints 		map[string]EndpointConfig 	`json:"endpoints"`
 }
 
 func (s *Sitecore) Enabled() bool {
-	return s.shouldDisplay()
+	if !s.env.HasFiles(sitecoreFileName) || !s.env.HasFiles(path.Join(sitecoreFolderName, userFileName)) {
+		return false
+	}
+
+	var userConfig = getUserConfig(s)
+
+	if (userConfig == nil) {
+		return false
+	}		
+
+	s.EndpointName = userConfig.getDefaultEndpoint()
+	
+	displayDefault := s.props.GetBool(properties.DisplayDefault, true)
+	
+	if !displayDefault && s.EndpointName == defaultEnpointName {
+		return false
+	} 
+
+	if endpoint := userConfig.getEndpoint(s.EndpointName); endpoint != nil && len(endpoint.Host) > 0 {
+		s.CmHost = endpoint.Host
+	}
+	
+	return true
 }
 
 func (s *Sitecore) Template() string {
-	return " {{ if .Cloud }}\uf65e{{ else }}\uf98a{{ end }} {{ .Environment }} "
+    return "{{ .EndpointName }} {{ if .CmHost }}({{ .CmHost }}){{ end }}"
 }
 
 func (s *Sitecore) Init(props properties.Properties, env platform.Environment) {
 	s.props = props
 	s.env = env
 }
 
-func (s *Sitecore) shouldDisplay() bool {
-	sitecoreDir, err := s.env.HasParentFilePath(".sitecore")
-	if err != nil {
-		return false
-	}
+func getUserConfig(s *Sitecore) *UserConfig {
```

Comment:
can we have this return an error on issues?

```suggestion
func getUserConfig(s *Sitecore) (*UserConfig, error) {
```

Reply:
yes, we can, but let me clarify one thing
is it necessary just to display an error message instead of hiding the whole segment? Or there is another aim of it?

example:
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/24988081/c159585b-a26e-4e6f-8741-9722a0317435)

Reply:
After update:
the method getUserConfig return error and this value is used for the if statement instead of userConfig value.

Reply:
> the method getUserConfig return error and this value is used for the if statement instead of userConfig value

@dorokhin-bohdan that's indeed was I was after 👍🏻